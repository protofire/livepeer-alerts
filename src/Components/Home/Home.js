import 'react-toastify/dist/ReactToastify.css'
import HomeCard from './HomeCard/HomeCard.js'
import React, { Component } from 'react'
import { toast, ToastContainer } from 'react-toastify'
import { withRouter } from 'react-router-dom'
import logdown from 'logdown'
const logger = logdown('Livepeer:Home')
logger.state.isEnabled = process.env.NODE_ENV !== 'production'

export class HomeComponent extends Component {
  state = {
    toastId: 1,
    displayMsg: null,
    error: false
  }

  onGetStartedBtnHandler = () => {
    logger.log('Fire event getStartedBtnHandler')
    this.props.history.push('/account')
  }

  onDemoBtnHandler = () => {
    logger.log('Fire event demoBtnHandler')
    this.props.history.push('/account/demo')
  }

  sendToast = (toastTime, callback) => {
    let time = 3500
    if (toastTime) {
      time = toastTime
    }
    let displayMsg = this.state.displayMsg
    if (!toast.isActive(this.state.toastId)) {
      if (this.state.error) {
        toast.error(displayMsg, {
          position: toast.POSITION.TOP_RIGHT,
          progressClassName: 'Toast-progress-bar',
          autoClose: time,
          toastId: this.state.toastId,
          onClose: callback
        })
      } else {
        toast.success(displayMsg, {
          position: toast.POSITION.TOP_RIGHT,
          progressClassName: 'Toast-progress-bar',
          autoClose: time,
          toastId: this.state.toastId,
          onClose: callback
        })
      }
      // TODO -- CHECK IF THERE IS ANOTHER WAY TO USE THIS, THIS IS FOR TESTING THAT THE TOAST IS CALLED
      if (this.props.toastOpenedHandlerTest) {
        this.props.toastOpenedHandlerTest(displayMsg)
      }
    }
  }

  componentDidMount() {
    logger.log('Fire event componentDidMount')
    // If we get redirected with an error msg, we should display
    if (this.props.location && this.props.location.state && this.props.location.state.error) {
      this.setState(
        {
          error: this.props.location.state.error,
          displayMsg: this.props.location.state.displayMsg
        },
        () => {
          this.sendToast()
        }
      )
    }
  }

  render() {
    return (
      <>
        <HomeCard onClick={this.onGetStartedBtnHandler} onDemoClick={this.onDemoBtnHandler} />
        <ToastContainer autoClose={2000} />
      </>
    )
  }
}
export default withRouter(HomeComponent)
