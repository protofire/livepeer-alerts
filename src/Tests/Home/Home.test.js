import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { HomeComponent } from '../../Components/Home/Home'
import * as texts from '../../Components/Common/UI/Texts/Texts'
import render from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('Renders home message', () => {
  it('Renders home message if user auth and match snapshot', () => {
    // Given
    let props = {
      web3: [],
      userData: {
        authenticated: true,
        reason: '',
        address: 123,
        currentNetwork: 123
      },
      render: false
    }
    let wrapper = render.create(<HomeComponent {...props} />)

    // When
    const tree = wrapper.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it('Renders error message if user not auth', () => {
    // Given
    const toastFn = jest.fn()
    let props = {
      web3: [],
      userData: {
        authenticated: false
      },
      render: true,
      location: {
        state: {
          displayMsg: texts.NO_PERMISSIONS,
          error: true
        }
      },
      toastOpenedHandlerTest: toastFn
    }

    // When
    let wrapper = shallow(<HomeComponent {...props} />)

    // Then
    expect(toastFn).toBeCalledWith(texts.NO_PERMISSIONS)
  })

  it('Renders error message if user does not have Metamask installed', () => {
    // Given
    const toastFn = jest.fn()
    let props = {
      web3: null,
      render: true,
      toastOpenedHandlerTest: toastFn,
      userData: {
        authenticated: false
      },
      location: {
        state: {
          displayMsg: texts.NO_WEB3,
          error: true
        }
      }
    }

    // When
    let wrapper = shallow(<HomeComponent {...props} />)

    // Then
    expect(toastFn).toBeCalledWith(texts.NO_WEB3)
  })
})
