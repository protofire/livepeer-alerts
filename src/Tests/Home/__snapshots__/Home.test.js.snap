// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders home message Renders home message if user auth and match snapshot 1`] = `
Array [
  <div
    className="MuiGrid-container-42 MuiGrid-align-items-xs-center-50 MuiGrid-justify-xs-center-59 GridContainer-grid-41 HomeCard-gridContainer-40"
  >
    <div
      className="MuiGrid-item-43 GridItem-grid-139 HomeCard-cardContainer-39"
    >
      <div
        className="Card-card-140 HomeCard-cardSignup-28"
      >
        <h2
          className="HomeCard-cardTitle-26"
        >
          Livepeer Notifications
        </h2>
        <div
          className="InfoArea-infoArea-157 "
        >
          <div
            className="InfoArea-iconWrapper-159 InfoArea-rose-165"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-171 InfoArea-icon-167"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M7.58 4.08L6.15 2.65C3.75 4.48 2.17 7.3 2.03 10.5h2c.15-2.65 1.51-4.97 3.55-6.42zm12.39 6.42h2c-.15-3.2-1.73-6.02-4.12-7.85l-1.42 1.43c2.02 1.45 3.39 3.77 3.54 6.42zM18 11c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2v-5zm-6 11c.14 0 .27-.01.4-.04.65-.14 1.18-.58 1.44-1.18.1-.24.15-.5.15-.78h-4c.01 1.1.9 2 2.01 2z"
              />
            </svg>
          </div>
          <div
            className="InfoArea-descriptionWrapper-168"
          >
            <h4
              className="InfoArea-title-169"
            >
              Pro-active alert notifications
            </h4>
            <p
              className="InfoArea-description-170"
            >
              We provide pro-active alert notifications that will help LPT token holders to be updated about how the delegates are performing in near real time
            </p>
          </div>
        </div>
        <div
          className="InfoArea-infoArea-157 "
        >
          <div
            className="InfoArea-iconWrapper-159 InfoArea-info-164"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-171 InfoArea-icon-167"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
          </div>
          <div
            className="InfoArea-descriptionWrapper-168"
          >
            <h4
              className="InfoArea-title-169"
            >
              You and Livepeer
            </h4>
            <p
              className="InfoArea-description-170"
            >
              You can see also a summary of your account information and your livepeer balance
            </p>
          </div>
        </div>
        <div
          className="InfoArea-infoArea-157 InfoArea-noCentered-158"
        >
          <div
            className="InfoArea-iconWrapper-159 InfoArea-primary-160"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-171 InfoArea-icon-167"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
              />
            </svg>
          </div>
          <div
            className="InfoArea-descriptionWrapper-168"
          >
            <h4
              className="InfoArea-title-169"
            >
              Open source
            </h4>
            <p
              className="InfoArea-description-170"
            >
              
            </p>
          </div>
        </div>
        <button
          className="MuiButtonBase-root-235 MuiButton-root-209 MuiButton-text-211 MuiButton-flat-214 RegularButton-button-180 RegularButton-lg-203 RegularButton-primary-182 RegularButton-round-205"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-210"
          >
            Get started
          </span>
          <span
            className="MuiTouchRipple-root-238"
          />
        </button>
        <button
          className="MuiButtonBase-root-235 MuiButton-root-209 MuiButton-text-211 MuiButton-flat-214 RegularButton-button-180 RegularButton-lg-203 RegularButton-info-183 RegularButton-round-205"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-210"
          >
            Demo
          </span>
          <span
            className="MuiTouchRipple-root-238"
          />
        </button>
      </div>
    </div>
  </div>,
  <div
    className="Toastify"
  />,
]
`;
